File: Structural/Lab9/C#/Builders/HouseBuilder.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Entities;

namespace C_.Builders
{
    public class HouseBuilder : EstateBuilder
    {        
        public override void Reset(){
            _estate = new House();
        }
        
        public HouseBuilder SetGarden(){
            var h = (House)_estate;
            h.Garden = true;
            return this;
        }

        public HouseBuilder SetPool(){
            var h = (House)_estate;
            h.Pool = true;
            return this;
        }
        public HouseBuilder SetGym(){
            var h = (House)_estate;
            h.Gym = true;
            return this;
        }
        public HouseBuilder SetHeatingType(string heating){
            var h = (House)_estate;
            h.HeatingType = heating;
            return this;
        }

    }
}

File: Structural/Lab8/Typescript/src/Builders/house_builder.ts
import { EstateBuilder } from './estate_builder';
import { House } from '../Entities/house';

export class HouseBuilder extends EstateBuilder {
    public reset(): void {
        this._estate = new House();
    }

    public setPool(): HouseBuilder {
        const a = this._estate as House;
        a.pool = true;
        return this;
    }

    public setGym(): HouseBuilder {
        const a = this._estate as House;
        a.gym = true;
        return this;
    }
    
    public setGarden(): HouseBuilder {
        const a = this._estate as House;
        a.garden = true;
        return this;
    }

    public setHeatingType(type: string): HouseBuilder {
        const a = this._estate as House;
        a.heatingType = type;
        return this;
    }
}
